openquake/commonlib/readinput.py

```diff
@@ -372,38 +372,41 @@ def get_source_models(oqparam, gsim_lt, source_model_lt, in_memory=True):
         oqparam.complex_fault_mesh_spacing,
         oqparam.width_of_mfd_bin,
         oqparam.area_source_discretization)
-    parser = nrml.SourceModelParser(converter)
+    psr = nrml.SourceModelParser(converter)
 
     # consider only the effective realizations
     for sm in source_model_lt.gen_source_models(gsim_lt):
-        fname = possibly_gunzip(
-            os.path.abspath(os.path.join(oqparam.base_path, sm.name)))
-        if in_memory:
-            apply_unc = source_model_lt.make_apply_uncertainties(sm.path)
-            try:
-                logging.info('Parsing %s', fname)
-                src_groups = parser.parse_src_groups(fname, apply_unc)
-            except ValueError as e:
-                if str(e) in ('Surface does not conform with Aki & '
-                              'Richards convention',
-                              'Edges points are not in the right order'):
-                    raise InvalidFile('''\
-    %s: %s. Probably you are using an obsolete model.
-    In that case you can fix the file with the command
-    python -m openquake.engine.tools.correct_complex_sources %s
-    ''' % (fname, e, fname))
-                else:
-                    raise
-        else:  # just collect the TRT models
-            smodel = nrml.read(fname).sourceModel
-            src_groups = []
-            if smodel[0].tag.endswith('sourceGroup'):  # NRML 0.5 format
-                for sg_node in smodel:
-                    sg = sourceconverter.SourceGroup(sg_node['tectonicRegion'])
-                    sg.sources = sg_node.nodes
-                    src_groups.append(sg)
-            else:  # NRML 0.4 format: smodel is a list of source nodes
-                src_groups.extend(sourceconverter.SourceGroup.collect(smodel))
+        src_groups = []
+        for name in sm.name.split():
+            fname = possibly_gunzip(
+                os.path.abspath(os.path.join(oqparam.base_path, name)))
+            if in_memory:
+                apply_unc = source_model_lt.make_apply_uncertainties(sm.path)
+                try:
+                    logging.info('Parsing %s', fname)
+                    src_groups.extend(psr.parse_src_groups(fname, apply_unc))
```

Comment:
There might be a case here where an error is generated. You are applying uncertainties separately to the various components (i.e. sources in the different files).  In case of an uncertainty applied to a specific source, that source might not be necessarily in the chunk of sources you are processing. Did I understand correctly?